### ISSUE TITLE
Code generation exits with error code 1 but is working properly

### ISSUE THREAD

**Issue by HerrNiklasRaab** [dart-lang/reflectable#197]
Hello,

The problem ist, that the command "flutter packages pub run build_runner build" exits with code 1. By removing the "extends Bloc<LoginEvent, LoginState>" from the LoginBloc class the command gets executed without failure. **In both cases, all files are generated properly and the whole project can be run without any problems.** 

In the login_bloc_test.dart the LoginBloc class is referenced but not used.

The LoginBloc class looks like following:

```dart
@inject
class LoginBloc extends Bloc<LoginEvent, LoginState> {
  UserService userService;

  @inject
  LoginBloc(this.userService);

  @override
  Stream<LoginState> mapEventToState(LoginEvent event) async* {
  }

  @override
  LoginState get initialState => LoginState();
}
```

Executing "flutter packages pub run build_runner build -v" results in the following output:

```
niklasraab@Niklass-MBP fazua_toolbox_app % flutter packages pub run build_runner build -v
[INFO] Generating build script...
[INFO] Generating build script completed, took 423ms

[INFO] BuildDefinition:Initializing inputs
[INFO] BuildDefinition:Reading cached asset graph...
[INFO] BuildDefinition:Reading cached asset graph completed, took 96ms

[INFO] BuildDefinition:Checking for updates since last build...
[INFO] BuildDefinition:Checking for updates since last build completed, took 1.0s

[INFO] Build:Running build...
[INFO] Heartbeat:1.0s elapsed, 0/1 actions completed.
[INFO] Heartbeat:5.8s elapsed, 0/1 actions completed.
[INFO] Heartbeat:12.4s elapsed, 0/1 actions completed.
[SEVERE] reflectable:reflectable on test/blocs/login_bloc_test.dart:
`reflectedTypeArguments` where an actual type argument (possibly nested) is void
[INFO] Heartbeat:14.4s elapsed, 0/1 actions completed.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
Attempt to generate code for an unsupported kind of type: void (VoidTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
Attempt to generate code for an unsupported kind of type: void (VoidTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: StreamSubscription<T> (InterfaceTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: S (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/future.dart:: Attempt to generate code for an unsupported kind of type: FutureOr<E> (InterfaceTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: Stream<E> (InterfaceTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:core/iterable.dart:: Attempt to generate code for an unsupported kind of type: Iterable<S> (InterfaceTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: S (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: S (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
Attempt to generate code for an unsupported kind of type: void (VoidTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: T (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
Attempt to generate code for an unsupported kind of type: void (VoidTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: EventSink<T> (InterfaceTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
Attempt to generate code for an unsupported kind of type: void (VoidTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
/bloc/lib/src/bloc.dart:11:28: Attempt to generate code for an unsupported kind of type: State (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
Attempt to generate code for an unsupported kind of type: void (VoidTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
dart:async/stream.dart:: Attempt to generate code for an unsupported kind of type: Stream<State> (InterfaceTypeImpl). Generating `dynamic`.
[WARNING] reflectable:reflectable on test/blocs/login_bloc_test.dart:
/bloc/lib/src/bloc.dart:11:21: Attempt to generate code for an unsupported kind of type: Event (TypeParameterTypeImpl). Generating `dynamic`.
[WARNING] Heartbeat:
No actions completed for 15.0s, waiting on:
  - reflectable:reflectable on test/blocs/login_bloc_test.dart

[INFO] Build:Running build completed, took 15.5s

[INFO] Build:Caching finalized dependency graph...
[INFO] Build:Caching finalized dependency graph completed, took 50ms

[SEVERE] Build:
Failed after 15.6s
pub finished with exit code 1

#0      throwToolExit (package:flutter_tools/src/base/common.dart:28:3)
#1      _DefaultPub.interactively (package:flutter_tools/src/dart/pub.dart:323:7)
<asynchronous suspension>
#2      PackagesPassthroughCommand.runCommand (package:flutter_tools/src/commands/packages.dart:269:15)
#3      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:615:18)
#4      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:73:64)
#5      _rootRunUnary (dart:async/zone.dart:1134:38)
#6      _CustomZone.runUnary (dart:async/zone.dart:1031:19)
#7      _FutureListener.handleValue (dart:async/future_impl.dart:139:18)
#8      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:680:45)
#9      Future._propagateToListeners (dart:async/future_impl.dart:709:32)
#10     Future._completeWithValue (dart:async/future_impl.dart:524:5)
#11     Future._asyncComplete.<anonymous closure> (dart:async/future_impl.dart:554:7)
#12     _rootRun (dart:async/zone.dart:1126:13)
#13     _CustomZone.run (dart:async/zone.dart:1023:19)
#14     _CustomZone.runGuarded (dart:async/zone.dart:925:7)
#15     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:965:23)
#16     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#17     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#18     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
#19     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:175:5)
```

If I am removing the "extends Bloc<LoginEvent, LoginState>" from the LoginBloc class the command get's executed without any problems.

**Comment by eernstg** [comment]
Thanks for reporting this, reflectable should definitely handle the case where `void` occurs as an actual type argument.

**Comment by HerrNiklasRaab** [comment]
Hello @eernstg,

is there a way to prevent exiting with an error. The files are generated as expected and everything in the project works as expected, but in our CI the build is stoping because this command exits with an error. 

Thanks,

Niklas

**Comment by eernstg** [comment]
Hello @HerrNiklasRaab, don't do `pub upgrade`! ;-)

Turns out that this causes some updates to be fetched in various packages, and this causes every code generation step to fail with a `[SEVERE]` problem, because an `InconsistentAnalysisException` is thrown. I'm asking on build, https://github.com/dart-lang/build/issues/2634, whether this is something that they know about.

**Comment by eernstg** [comment]
Issue #198 keeps track of the `InconsistentAnalysisException`.

**Comment by eernstg** [comment]
You could try out the branch 'support_reified_void_feb20'.

**Comment by eernstg** [comment]
Landed #199 as reflectable 2.2.3.

**Comment by HerrNiklasRaab** [comment]
Can confirm, is fixed! Thank you.
